cmflib/cmfquery.py

```diff
@@ -14,414 +14,670 @@
 
-        return df
+        Args:
+            stage_name: Stage name.
+        Returns:
+            Data frame with all executions associated with the given stage.
 
-    def get_artifact_df(self, node):
-        d = {"id": node.id, "type": self.store.get_artifact_types_by_id([node.type_id])[0].name, "uri": node.uri,
-             "name": node.name, "create_time_since_epoch": node.create_time_since_epoch,
-             "last_update_time_since_epoch": node.last_update_time_since_epoch}
-        for k, v in node.properties.items():
-            if v.HasField('string_value'):
-                d[k] = v.string_value
-            elif v.HasField('int_value'):
-                d[k] = v.int_value
-            else:
-                d[k] = v.double_value
-        for k, v in node.custom_properties.items():
-            if v.HasField('string_value'):
-                d[k] = v.string_value
-            elif v.HasField('int_value'):
-                d[k] = v.int_value
-            else:
-                d[k] = v.double_value
-        df = pd.DataFrame(d, index=[0, ])
+        TODO: Multiple stages with the same name? This method collects executions from all such stages. Also, see
+              "get_all_exe_in_stage"
+        """
+        df = pd.DataFrame()
+        for pipeline in self._get_pipelines():
+            for stage in self._get_stages(pipeline.id):
+                if stage.name == stage_name:
+                    for execution in self._get_executions(stage.id):
+                        ex_as_df: pd.DataFrame = self._transform_to_dataframe(
+                            execution, {"id": execution.id, "name": execution.name}
+                        )
+                        df = pd.concat([df, ex_as_df], sort=True, ignore_index=True)
         return df
 
-    def get_all_artifacts(self) -> t.List:
-        artifact_list = []
-        artifacts = self.store.get_artifacts()
-        for art in artifacts:
-            name = art.name
-            artifact_list.append(name)
-        return artifact_list
-
-    def get_artifact(self, name: str):
-        artifact = None
-        artifacts = self.store.get_artifacts()
-        for art in artifacts:
-            if art.name == name:
-                artifact = art
-                break
-        return self.get_artifact_df(artifact)
-
-    def get_all_artifacts_for_execution(self, execution_id: int) -> pd.DataFrame: # change here
+    def get_artifact_df(self, artifact: mlpb.Artifact, d: t.Optional[t.Dict] = None) -> pd.DataFrame:
+        """Return artifact's data frame representation.
+
+        Args:
+            artifact: MLMD entity representing artifact.
+            d: Optional initial content for data frame.
+        Returns:
+            A data frame with the single row containing attributes of this artifact.
+
+        TODO: (sergey) there are no "public" methods that return `mlpb.Artifact`.
+        TODO: (sergey) what's the  difference between this method and `get_artifact`?
```

Comment:
get_artifact takes a string name and returns a df of the artifact.
get_artifact_df takes an artifact and returns its df